### ISSUE TITLE
[Help]: nearest_grid is painfully slow

### ISSUE THREAD

**Issue by SteffanDavies** [gmtsar/gmtsar#618]
@mobigroup @Xiaohua-Eric-Xu GMTSAR nearest grid is taking one hour to complete for each interferogram of 2 merged subswaths.

Note that almost no NaNs in mask. 

![image](https://user-images.githubusercontent.com/50405969/211944233-fa688b75-61a0-4228-a798-b5b04d837f88.png)

![image](https://user-images.githubusercontent.com/50405969/211944800-ff7d48fd-d2dc-4750-90a2-62b90b1a340c.png)

There are no significant water bodies in the area. Any idea what is happening?

**Comment by SteffanDavies** [comment]
I am considering splitting the snaphu_interp.csh file into a nearest_grid containing script and a snaphu containing script. This will allow running the nearest_grid part in parallel for multiple interferograms and then running the unwrapping on each interferogram separately in parallel using tiling.

**Comment by Xiaohua-Eric-Xu** [comment]
I am not sure why it's running so long. If the interpolation is done with your mask_def.grd only, it should finish in a few minutes. As answered in another issue #612, you could replace the nearest_grid part with other interpolation methods. I used Matlab griddata once and the solution are identical.

**Comment by mobigroup** [comment]
@Xiaohua-Eric-Xu But what is the reason to have so problematic nearest_grid tool and recommend to use other interpolation tools for the every user who asks about the problem? That requires to change the processing pipeline to add user-defined interpolation.

**Comment by Xiaohua-Eric-Xu** [comment]
The tool was built to interpolate through scatters to simulate the scenario in this paper (Sparse Two-Dimensional Phase Unwrapping Using Regular Grid Methods). Since we could not guarantee all users have access to Matlab, so I wrote a little program to do that. It's something that works but not idea for all scenarios. In fact, the histogram approach (I think that's what Matlab uses) takes much longer to run for large volume of scattered data compared to nearest_grid, but I would not call the histogram approach problematic. 

![image](https://user-images.githubusercontent.com/16844111/212001231-31adbd38-f682-448f-a80f-531e40b7e5fe.png)

**Comment by SteffanDavies** [comment]
@Xiaohua-Eric-Xu @Xiaohua-Eric-Xu I understand the problem now.

Merged scenes create NaN is natural subswath shift. See here:

![image](https://user-images.githubusercontent.com/50405969/212086328-3e041fbb-964f-4269-a696-d6450fae98b5.png)
 
The issue can be solved by unwrapping only within lines containing data on both SS, making nearest_grid instant.

In this case, unwrapping within 464/12195 (ymin/ymax).
